src/PrestaShopBundle/Controller/Admin/Improve/Modules/AddonsStoreController.php

```diff
@@ -35,28 +36,37 @@
 class AddonsStoreController extends FrameworkBundleAdminController
 {
     /**
-     * @var string the controller name for routing
-     */
-    const CONTROLLER_NAME = 'AdminAddonsCatalog';
-
-    /**
+     *
+     * @AdminSecurity("is_granted(['read'], request.get('_legacy_controller'))")
+     *
      * @param Request $request
      *
      * @return \Symfony\Component\HttpFoundation\Response
      */
     public function indexAction(Request $request)
     {
+        $pageContent = @file_get_contents($this->getAddonsUrl($request));
+
+        if (!$pageContent) {
```

Comment:
I added this to handle the usecase "Addons returns nothing". It can be improved.

Reply:
I agree, but this is a first start of adding business logic here :/

How about extracting this in its own service (this can be reused for theme page too)?

I suggest to address the fix of Addons unavailability in another pull request :+1: